src/sth_database.js

```diff
@@ -812,6 +852,84 @@
     });
   }
 
+  /**
+   * Generates a hash based on an input and a maximum number of bytes
+   * @param input The input to generate the hash from
+   * @param limit The maximum number of bytes of the hash
+   */
+  function generateHash(input, limit) {
+    var shasum = crypto.createHash('sha512');
+    shasum.update(input);
+    var hash = shasum.digest('hex');
+    if (limit) {
+      hash = hash.substr(0, limit);
+    }
+    return hash;
+  }
+
+  /**
+   * Stores the collection name (hash) in the database
+   * @param params The params used to generate the collection name (hash)
+   * @param hash The collection name (hash)
+   * @param callback A callback function
+   */
+  function storeCollectionHash(params, isAggregated, hash, callback) {
+    getCollection({
+      service: params.service,
+      collection: sthConfig.COLLECTION_PREFIX + 'collection_names'
+    }, false, true, false, function(err, collection) {
+      if (err) {
+        return callback(err);
+      }
+      // 2 updates operations are needed since MongoDB currently does not support the possibility
+      //  to address the same field in a $set operation as a $setOnInsert operation
+      collection.update(
+        {
+          _id: hash
+        },
+        {
+          '$setOnInsert': {
+            dataModel: sthConfig.DATA_MODEL,
+            isAggregated: isAggregated,
```

Comment:
It is :) I only included it to ease and to speed up queries :) Otherwise, users would have to use text searching in the _id property of the collection which maps hashes to service path, entity, etc., vales... This will force us to create a text index on that property (see http://docs.mongodb.org/manual/reference/operator/query/text/) which apart of consuming a namespace (this is not important :D) can end up consuming a lot of disk and memory (this is the counterpart of text indexes, and the reason why they are not recommended if not forced to really use them :))

What do you think?